syntax = "proto2";

package trader;

// Security (crypto currency) price at a specific time.
message PriceRecord {
  // UNIX timestamp (in seconds).
  optional int32 timestamp_sec = 1;
  // Security (crypto currency) price (e.g. BTC/USD).
  optional float price = 2;
  // Traded security (crypto currency) volume.
  optional float volume = 3;
}

// Open, high, low, and close prices for specific time interval.
// The duration of the time interval is assumed implicitly.
message OhlcTick {
  // UNIX timestamp (in seconds) of the start of the time interval.
  optional int32 timestamp_sec = 1;
  // Opening security (crypto currency) price at the start of the time interval.
  optional float open = 2;
  // Highest security (crypto currency) price during the time interval.
  optional float high = 3;
  // Lowest security (crypto currency) price during the time interval.
  optional float low = 4;
  // Closing security (crypto currency) price at the end of the time interval.
  optional float close = 5;
  // Total traded volume during the time interval.
  optional float volume = 6;
}
